services/manager/internal/service/create_desired_state.go

```diff
@@ -83,13 +83,21 @@ func createK8sClustersFromManifest(from *manifest.Manifest, into *store.Config)
 	//    catching newly created or existing (updated).
 	desiredClusters := make(map[string]struct{})
 	for _, cluster := range from.Kubernetes.Clusters {
+		useInstallationProxy := &spec.InstallationProxy{}
+
+		if cluster.InstallationProxy != nil {
+			useInstallationProxy.Mode = cluster.InstallationProxy.Mode
+			useInstallationProxy.Endpoint = cluster.InstallationProxy.Endpoint
+		}
+
```

Comment:
If the proxy is not set this will result in an empty `*spec.InstallationProxy` however in builder you are by default setting the `default` setting and after the workflow in builder finishes the current state and desired state will diverge as desired state as created in this function will have
```go
&spec.InstallationProxy{
Mode: ""
Endpoing : ""
}
```

and the current state after the workflow finishes

```go
&spec.InstallationProxy{
Mode: "default"
Endpoing : ""
}
```